internal/fetchclient/fetchclient.go

```diff
@@ -115,6 +118,52 @@ func (c *Client) fetchDartFlutter(ctx context.Context, name string) (string, err
 	return data.Latest.Version, nil
 }
 
+func (c *Client) fetchCrate(ctx context.Context, name string) (string, error) {
+	request, err := http.NewRequestWithContext(
+		ctx,
+		http.MethodGet,
+		fmt.Sprintf("%s/crates/%s", cratesURL, strings.TrimPrefix(name, "/")),
+		nil,
+	)
+	if err != nil {
+		return "", err
+	}
+	// See https://github.com/bufbuild/plugins/issues/252 for more information.
+	// We must be careful with this API and respect the crawling policy.
+	request.Header.Set("User-Agent", "bufbuild (github.com/bufbuild/plugins)")
+	response, err := c.httpClient.Do(request)
+	if err != nil {
+		return "", err
+	}
+	defer response.Body.Close()
+
+	var data struct {
+		Versions []struct {
+			Yanked bool   `json:"yanked"`
+			Num    string `json:"num"`
+		} `json:"versions"`
+	}
+	if err := json.NewDecoder(response.Body).Decode(&data); err != nil {
+		return "", err
+	}
+	var versions []string
+	for _, version := range data.Versions {
+		if version.Yanked {
+			// A yanked version a is a published crate's version that has been removed
+			// from the server's index.
```

Comment:
If we encounter a crate that was the latest release and then later "yanked", what will be our solution on the BSR? Will we need the ability to disable a plugin so that newer code that doesn't specify a version and remote packages builds with the last stable release?

Reply:
The crate is still available for download, so technically it's still available even upstream.

I think the best we can do is mark it deprecated, and that'll be a manual process.

We could automate this and maybe create a GH issue if we detect this, which gets assigned to a person. But I'm a bit worried about the ongoing maintenance cost of trying to mirror "deprecated" plugin versions.

Reply:
Tbh I don't have a good answer. Suggestions welcome.

If we can't be consistent, across the ecosystems, then maybe the best thing is to do nothing and leave it to the user?